python/cudf/cudf/core/column/categorical.py

```diff
@@ -582,7 +595,26 @@ def is_monotonic_decreasing(self):
         return self._is_monotonic_decreasing
 
     def as_categorical_column(self, dtype, **kwargs):
-        return self
+        if isinstance(dtype, str) and dtype == "category":
+            return self
+        if isinstance(
+            dtype, (cudf.core.dtypes.CategoricalDtype, pd.CategoricalDtype)
+        ):
+            if (dtype.categories is None) and (dtype.ordered is None):
```

Comment:
You can combine these two conditions into the above if to prevent needing nesting. Python will evaluate the conditions in order and short circuit if the first one is `False`.